objetos/funcionarios/Funcionario.java

```diff
@@ -65,6 +70,28 @@ public boolean setNivelCargo(String nivelCargo) {
         }
     }
 
+    //Promover - apenas salário
+    public void promover(double valorAdicionalSalario) {
+        this.valorAdicionalSalario = valorAdicionalSalario;
+        this.setSalario(this.salario + valorAdicionalSalario);
+    }
+
+    //Promover - Cargo com aumento salarial
+    public void promover(String nivelCargo, double valorAdicionalSalario) {
+        this.setNivelCargo(nivelCargo);
+        this.valorAdicionalSalario = valorAdicionalSalario;
+
+        if (nivelCargo.equals("AVANCADO") || nivelCargo.equals("EXPERIENTE")) {
+            this.salario += valorAdicionalSalario;
+        } else {
+            System.out.println("Não há promoção para o cargo de nível INICIANTE.");
+        }
+    }
+
+    public void promover(Funcionario funcionario,String nivelCargo) {
```

Comment:
No método promover você está passando como parâmetro o nível do cargo. 
Nossa sugestão foi não passar mais o parâmetro e deixar essa alteração de uma forma lógica.
Sendo assim ele receberia só por parâmetro o novo salário que subistituirá o salário antigo.
Também vejo que não precisamos do funcionário como parâmetro, pois ele é o this.

Reply:
feito